lib/blueprinter/v2.rb

```diff
@@ -0,0 +1,116 @@
+# frozen_string_literal: true
+
+require 'blueprinter/v2/dsl'
+require 'blueprinter/v2/options'
+require 'blueprinter/v2/reflection'
+require 'blueprinter/v2/view_builder'
+
+module Blueprinter
+  # Base class for V2 Blueprints
+  class V2
+    extend DSL
+    extend Reflection
+
+    class << self
+      # Options set on this Blueprint
+      attr_accessor :options
+      # Extensions set on this Blueprint
+      attr_accessor :extensions
+      # @api private The fully-qualified name, e.g. "MyBlueprint", or "MyBlueprint.foo.bar"
+      attr_accessor :blueprint_name
+      # @api private
+      attr_accessor :views, :fields, :excludes, :partials, :used_partials, :eval_mutex
+    end
+
+    self.views = ViewBuilder.new(self)
+    self.fields = {}
+    self.excludes = []
+    self.partials = {}
+    self.used_partials = []
+    self.extensions = []
+    self.options = Options.new(DEFAULT_OPTIONS)
```

Comment:
**nb**: I'd expect `DEFAULT_OPTIONS` to be encapsulated solely within `Options`, and that simply calling `Options.new` would initialize it with the defaults.